src/fortune.cpp

```diff
@@ -99,9 +99,9 @@ void budget::status_fortunes(budget::writer& w, bool short_view){
                     contents.push_back({"", to_string(fortune.check_date), to_string(fortune.amount), "", "", "", "", "", "::edit::fortunes::" + budget::to_string(fortune.id)});
                 }
             } else if (i == 1) {
-                auto diff = fortune.amount - previous;
-                auto d    = fortune.check_date - previous_date;
-                auto avg  = diff / d;
+                money diff = fortune.amount - previous;
+                long d     = fortune.check_date - previous_date;
+                money avg  = diff / d;
```

Comment:
What's clang saying on these lines?

Reply:
```


budgetwarrior/src/fortune.cpp:104:34: error: use of overloaded operator '/' is ambiguous (with operand types
      'budget::money' and 'long long')
                auto avg  = diff / d;
                            ~~~~ ^ ~
budgetwarrior/include/money.hpp:136:11: note: candidate function
    money operator/(int factor) const {
          ^
budgetwarrior/include/money.hpp:142:11: note: candidate function
    money operator/(long factor) const {
          ^
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, long long)
                auto avg  = diff / d;
                                 ^
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(float, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(double, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(long double, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, float)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, double)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, long double)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, int)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, __int128)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, unsigned int)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, unsigned long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, unsigned long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(int, unsigned __int128)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(long, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(long long, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(__int128, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(unsigned int, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(unsigned long, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(unsigned long long, long long)
budgetwarrior/src/fortune.cpp:104:34: note: built-in candidate operator/(unsigned __int128, long long)
```

Reply:
The issue is the `value_type` of date is unsigned short.  Another fix would be to case the date diff to `long` to resolve the overload ambiguity. 

```
auto diff = fortune.amount - previous;                                                                                          
auto d     = static_cast<long>(fortune.check_date - previous_date);                                                             
auto avg  = diff / d;  
```

It's up to you. I've learned that `auto` is over-used at times. Sometimes it's clearer and better to just specify the type.

Reply:
Thanks for sharing the details. I agree with your fix. I will do some improvements on the operators themselves later.